Userland/Libraries/LibWeb/Infra/Lists.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2023, Kenneth Myhra <kennethmyhra@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+
+namespace Web::Infra {
+
+// https://infra.spec.whatwg.org/#list-clone
+template<typename Key, typename Value>
+ErrorOr<HashMap<Key, Value>> clone_list(HashMap<Key, Value> const& entry_list)
```

Comment:
An infra "list" doesn't appear to be a key/value mapping, this only works in the spec because "entry lists" are lists of key/value tuples. I'd suggest we skip this commit and add fallible clone functions to Vector/HashMap/etc. instead (possibly just using a copy ctor in the LibWeb code that needs it, with a FIXME).

Reply:
Dropped this and added a `clone()` function to `HashMap` which does a shallow clone. Was going back and forth with myself if I should call it `shallow_clone` or `clone`, ended up with `clone`.